File: 409/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int [] H = new int[26+26];
        int N = s.length();
        for(int i=0; i<N; i++){
            char c = s.charAt(i);
            if(c>='A' && c<='Z') H[c-'A']++;
            if(c>='a' && c<='z') H[c-'a'+26]++;
        }
        int odd=0;
        int ans=0;
        for(int i=0; i<26+26; i++){
            if(H[i]%2==1) odd++;
            if(H[i]>=2) ans += H[i]/2*2;
        }
        if(odd>0) ans++;
        return ans;
    }
}

File: 409/longest-palindrome.c
int longestPalindrome(char * s){
    int ans=0;
    int H1[26]={};
    int H2[26]={};
    while(*s!=0){
        if(*s>='a'&& *s<='z') H1[*s-'a']++;
        else H2[*s-'A']++;
        s++;
    }
    int odd=0;
    for(int i=0; i<26; i++){
        if(H1[i]%2==1) odd=1;
        if(H2[i]%2==1) odd=1;
        ans += H1[i]/2*2;
        ans += H2[i]/2*2;
    }
    if(odd==1) ans++;
    return ans;
}//case 26/95: "AAAAAA"
